docs/unified-test-documentation/dasharo-security/200-tpm-support.md

```diff
@@ -404,3 +404,188 @@ Example output:
+
+    ```bash
+    sudo tpm2_sign -c key.ctx -o sig.rssa secret.data
+    ```
+
+1. Run the following command to verify the signed file:
+
+    ```bash
+    sudo tpm2_verifysignature -c key.ctx -s sig.rssa -m secret.data
+    ```
+
+**Expected result**
+
+The output of the last command shouldn't display any warnings and errors.
+
+## TPM014.001 Encryption and Decryption of the file (Ubuntu 22.04)
+
+**Test description**
+
+This test aims to verify that the TPM supports the encryption and decryption of
+the file.
+
+**Test configuration data**
+
+1. `FIRMWARE` = Dasharo
+1. `OPERATING_SYSTEM` = Ubuntu 22.04
+
+**Test setup**
+
+1. Proceed with the
+    [Test cases common documentation](#test-cases-common-documentation) section.
+1. Install the `tpm2-tools` package: `sudo apt install tpm2-tools`.
+
+**Test steps**
+
+1. Power on the DUT.
+1. Boot into the system.
+1. Log into the system by using the proper login and password.
+1. Open the terminal and run the following set of commands to generate the
+   required keys and files:
+
+    ```bash
+    sudo tpm2_clear
+    sudo tpm2_createprimary -c primary_key.ctx
+    sudo tpm2_create -u key.pub -r key.priv -C primary_key.ctx
+    sudo tpm2_load -C primary_key.ctx -u key.pub -r key.priv -c key.ctx
+    echo "my secret" > secret.data
+    ```
+
+1. Run the following command to encrypt the file:
+
+    ```bash
+    sudo tpm2_encryptdecrypt -c key.ctx -o secret.enc secret.data
+    ```
+
+1. Run the following command to decrypt the file:
+
+    ```bash
+    sudo tpm2_encryptdecrypt -d -c key.ctx -o secret.dec secret.enc
+    ```
+
+1. Run the following command to check the file content:
+
+    ```bash
+    cat secret.dec
+    ```
+
+**Expected result**
+
+The output of the last command should display the content of the `secret.data`
+file, in this case, it should be `my secret`.
+
+## TPM015.001 Hashing the file (Ubuntu 22.04)
+
+**Test description**
+
+This test aims to verify that the TPM supports file hashing.
+
+**Test configuration data**
+
+1. `FIRMWARE` = Dasharo
+1. `OPERATING_SYSTEM` = Ubuntu 22.04
+
+**Test setup**
+
+1. Proceed with the
+    [Test cases common documentation](#test-cases-common-documentation) section.
+1. Install the `tpm2-tools` package: `sudo apt install tpm2-tools`.
+
+**Test steps**
+
+1. Power on the DUT.
+1. Boot into the system.
+1. Log into the system by using the proper login and password.
+1. Open the terminal and run the following set of commands to hash the file:
+
+    ```bash
+    sudo tpm2_clear
+    echo "my secret" > secret.data
+    sudo tpm2_hash -o hash.out -t ticket.out secret.data
+    ```
+
+**Expected result**
+
+1. The output of the last command shouldn't display any warnings and errors.
+1. Files `hash.out` and `ticket.out` should be correctly created and shouldn't
+   be empty.
+
+## TPM016.001 Performing HMAC operation on the file (Ubuntu 22.04)
+
+**Test description**
+
+This test aims to verify that the TPM supports HMAC operation.
```

Comment:
Could we briefly describe what is the HMAC operation?

Reply:
Added